### PR TITLE
Update public API

### PR DESCRIPTION
**Pull request by mannprerak2**
- Updated `Config`'s class members to use public getter, private variable pattern.
- Replaced ConfigError with FormatException
- Refactored all shared global variables to data.dart
- Send test coverage to coveralls in travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,14 +12,20 @@ cache:
...
     - $HOME/.pub-cache
 before_install:
-  # install libclang-dev
+  # Install libclang-dev.
   - sudo add-apt-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-10 main"
   - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
   - sudo apt-get update
   - sudo apt-get install clang-10 libclang-10-dev
-  # use clang from installed llvm
+  # Use clang from installed llvm
```
Comment (dcharkes) [inline .travis.yml:20]: .

File: .travis.yml [hunk 2]
```diff
@@ -12,14 +12,20 @@ cache:
...
   - export PATH="/usr/lib/llvm-10/bin:$PATH"
 
 before_script:
   - 'pub run ffigen:setup'
   - cd test/native_test && dart build_test_dylib.dart && cd ../..
+
+# Generate and upload coverage report
```
Comment (dcharkes) [inline .travis.yml:27]: .

File: lib/src/header_parser/parser.dart [hunk 3]
```diff
@@ -60,7 +55,12 @@ void initParser(Config c) {
       strings.ffigenFolderName,
       strings.dylibFileName,
     );
-    clang = clang_types.Clang(DynamicLibrary.open(fullDylibPath));
+
+    // Initialise global variables.
+    initialiseGlobals(
```
Comment (dcharkes) [inline lib/src/header_parser/parser.dart:60]: Please check whether you use UK (initialise) or US (initialize) spelling consistently in the repo.

File: lib/src/header_parser/utils.dart [hunk 4]
```diff
@@ -322,3 +322,60 @@ class IncrementalNamer {
...
+
+  bool isSeenStruc(String originalName) {
+    return _structs.containsKey(originalName);
+  }
+
+  void addStrucToSeen(String originalName, Struc struc) {
+    _structs[originalName] = struc;
+  }
+
+  Struc getSeenStruc(String originalName) {
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:342]: isSeenStruct

File: lib/src/header_parser/utils.dart [hunk 5]
```diff
@@ -322,3 +322,60 @@ class IncrementalNamer {
...
+  final Map<String, Struc> _structs = {};
+  final Map<String, Func> _functions = {};
+  final Map<String, EnumClass> _enumClass = {};
+  final Map<String, String> _macros = {};
+
+  bool isSeenStruc(String originalName) {
+    return _structs.containsKey(originalName);
+  }
+
+  void addStrucToSeen(String originalName, Struc struc) {
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:338]: addStructToSeen

File: lib/src/header_parser/utils.dart [hunk 6]
```diff
@@ -322,3 +322,60 @@ class IncrementalNamer {
     return '${base}_$i';
   }
 }
+
+/// Tracks if a binding is 'seen' or not.
+class SeenTracker {
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:327]: Maybe call it a `BindingsIndex`?

File: .travis.yml (hunk 2 above)
Comment (mannprerak2) [reply .travis.yml]: I have removed coverage report generation from travis

File: lib/src/header_parser/parser.dart (hunk 3 above)
Comment (mannprerak2) [reply lib/src/header_parser/parser.dart:60]: Updated it to US spelling. (also used in sqlite, libclang)

File: lib/src/header_parser/utils.dart [hunk 7]
```diff
@@ -322,3 +322,60 @@ class IncrementalNamer {
     return '${base}_$i';
   }
 }
+
+/// Tracks if a binding is 'seen' or not.
+class BindingsIndex {
+  // Stores binding names already scene. Mp key is same as their original name.
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:328]: scene --> seen

Comment (dcharkes) [inline lib/src/header_parser/utils.dart:328]: Mp -> Map

**Pull request merged**